x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -444,20 +444,22 @@ export class SessionIndex {
    * Trigger a removal of any outdated session values.
    */
   async cleanUp() {
-    this.options.logger.debug(`Running cleanup routine.`);
+    const { auditLogger, elasticsearchClient, logger } = this.options;
+    logger.debug(`Running cleanup routine.`);
 
+    let error: Error | undefined;
+    let haveDeletedSessions = false;
     try {
       for await (const sessionValues of this.getSessionValuesInBatches()) {
         const operations: Array<Required<Pick<BulkOperationContainer, 'delete'>>> = [];
         sessionValues.forEach(({ _id, _source }) => {
           const { usernameHash, provider } = _source!;
-          this.options.auditLogger.log(
-            sessionCleanupEvent({ sessionId: _id, usernameHash, provider })
-          );
+          auditLogger.log(sessionCleanupEvent({ sessionId: _id, usernameHash, provider }));
           operations.push({ delete: { _id } });
         });
         if (operations.length > 0) {
-          const bulkResponse = await this.options.elasticsearchClient.bulk(
+          haveDeletedSessions = true;
```

Comment:
Technically speaking we haven't actually deleted anything at this point. Is it worth being stricter here and only marking sessions as having been touched after the bulk delete operation?

Reply:
I probably could have clarified this in the comments:

My reasoning for doing it here is that I'm not 100% sure if a failed `bulk` operation could have resulted in _some_ of the sessions getting deleted. I figured it's safest to trigger an index refresh anyway, but I'm not sure how much of a practical impact this has tbh. I suppose `haveAttemptedToDeleteSessions` would have been a more accurate variable name but that sounds like a mouthful 😅

Reply:
Changed here: f4be0217d0bd91bbb4d7791d3c12aa6d530bded8

Let me know what you think!